### DISCUSSION TITLE
type-of on values instead of just symbols?

### DISCUSSION THREAD

**Discussion by Inc0n** [coalton-lang/coalton#768] [Help!]
I am learning Coalton, and I found little resources to help me get started, given my limited exposure to haskell.

And I am thinking about a `type-of` that can run on values to explore the type system of `coalton` would be friendly. Which could be similar to `typed/racket`, which prints out the type of the value entered on repl.

**Comment by aijony** [comment]
You will be able to do something like that with:
https://github.com/coalton-lang/coalton/pull/682

It has a few subtle issues I haven't ironed out, but I will hopefully get to it soon.
